src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
+ 
+  (define subexprs-fn (eval-progs-real subexprs-list ctx-list))
+ 
+  (define error-count-hash
+    (make-hash (map (lambda (x) (cons x '())) (cons #f subexprs-list))))
+ 
+  (define (mark-erroneous! expr pt)
+    (hash-update! error-count-hash expr (lambda (x) (set-add x pt))))
+  
+  (for ([(pt _) (in-pcontext pctx)])
+    (define exacts (apply subexprs-fn pt))
+    (define exacts-hash
+      (make-immutable-hash (map cons subexprs-list exacts)))
+ 
+    (define uflow-hash (make-hash (map (lambda (x) (cons x #f)) subexprs-list)))
+    (define (underflow? subexpr) (hash-ref uflow-hash subexpr))
+    (define oflow-hash (make-hash (map (lambda (x) (cons x #f)) subexprs-list)))
+    (define (overflow? subexpr) (hash-ref oflow-hash subexpr))
+ 
+    (unless
+        (ormap identity
+               (for/list ([subexpr subexprs-list])
+                 (define subexpr-val (hash-ref exacts-hash subexpr))
+
+                 (cond
+                   ; NOTE need better way to see if a calculation underflowed
+                   [(fl= subexpr-val 0.0)
+                    (hash-set! uflow-hash subexpr #t)]
+                   [(infinite? subexpr-val)
+                    (hash-set! oflow-hash subexpr #t)])
+                 
+                 (match subexpr
+                   [(list '+.f64 larg rarg)
+                    #:when (or (is-inexact? larg) (is-inexact? rarg))
+                    (define larg-val (hash-ref exacts-hash larg))
+                    (define rarg-val (hash-ref exacts-hash rarg))
+                    (define x+y (+ larg-val rarg-val))
+                    (define cond-x (abs (/ larg-val x+y)))
+                    (define cond-y (abs (/ rarg-val x+y)))
+                    
+                    (cond
+                      ; Both R(x + y) and R(x) + R(y) underflow
+                      ; This causes the condition number to jump up, with no real error
+                      [(and (= x+y 0.0) (underflow? subexpr)) #f]
+
+                      ; nan rescue:
+                      ; R(+-inf) + R(-+inf) = nan, but should actually get inf 
+                      [(and (infinite? larg-val)
+                            (infinite? rarg-val)
+                            (not (same-sign? larg-val rarg-val))
+                            (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+
+                      ; inf rescue:
+                      ; R(inf) + y = non inf value (inf rescue)
+                      [(and (or (infinite? larg-val)
+                                (infinite? rarg-val))
+                            (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; High condition number:
+                      ; CN(+, x, y) = |x / x + y| 
+                      [(or (> cond-x 1e2) (> cond-y 1e2))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
```

Comment:
I think if we move the mark-erroneous definition into the loop and have it set a boolean we can avoid nesting everyth8ng into an unless or.